### DISCUSSION TITLE
Confluent.Kafka.ConsumeException: Unknown broker error

### DISCUSSION THREAD

**Discussion by Atallah-Hammad** [edenhill/librdkafka#3769] [Q&A]
Description
===========
I'm using [azure-event-hubs-for-kafka], and sometimes getting this error when consume=>
Confluent.Kafka.ConsumeException: Unknown broker error
at Confluent.Kafka.Consumer`2.Consume(Int32 millisecondsTimeout)

How to reproduce
================
Using .NET 5 class library with Confluent.Kafka 1.8.2, and this error occurs also when using 1.6.2.

Is BrokerVersionFallback config property can help here?

Has it worked previously?
=========================
Sometimes getting this error, also in the previous versions.

Checklist
=========
confluent-kafka-dotnet 1.82
ConsumerConfig { BootstrapServers = KafkaCredentials.BrokerList, SecurityProtocol = SecurityProtocol.SaslSsl, SocketTimeoutMs = 60000, SessionTimeoutMs = 30000, SaslMechanism = SaslMechanism.Plain, SaslUsername = "$ConnectionString", SaslPassword = KafkaCredentials.ConnectionString, GroupId = consumerGroup, AutoOffsetReset = AutoOffsetReset.Latest }

**Comment by edenhill** [comment]
Please enable debugging by setting `Debug = "all"` in your ConsumerConfig and provide the full log output.

**Reply by Atallah-Hammad** [reply]
Thanks for the answer.
- There are any side effects of running it in Debug mode?
- What do you mean by full log? I will catch the ConsumeException, is that enough?

**Reply by edenhill** [reply]
1. Nope, should be fine.
2. Logs will be printed to your console / stdout. Collect them all and upload here.

**Reply by Atallah-Hammad** [reply]
I do that, and I have many logs, but in my case, this error does not always appear.
What we can do instead?

**Reply by Atallah-Hammad** [reply]
Any suggestions, please?

**Reply by edenhill** [reply]
Please provide logs.

**Reply by Atallah-Hammad** [reply]
[logs.txt](https://github.com/edenhill/librdkafka/files/8357054/logs.txt)
I copied what was printed on the console, and provided them in a text file.

**Comment by Atallah-Hammad** [comment]
[logs.txt](https://github.com/edenhill/librdkafka/files/8357053/logs.txt)
I copied what was printed on console, and provided them in a text file.

**Reply by edenhill** [reply]
I can't see any "Unknown broker error" in that log

**Reply by Atallah-Hammad** [reply]
Hi, I added a new answer.
Any suggestions, please?

**Comment by Atallah-Hammad** [comment]
Hi, I stored logs all day, and have no "Unknown broker error" in that logs, although it was caught in my error logs.
[ProdLogs.txt](https://github.com/edenhill/librdkafka/files/8372666/ProdLogs.txt).

And this is the screenshot for the error in my DB:
![image](https://user-images.githubusercontent.com/59364267/160642040-370d6658-e5d3-4146-9ca4-5ec7c57a7dcc.png)

And this is the Message content:
Message=Unknown broker error
ErrorReason=Unknown broker error
ConsumeExceptionJson={"ConsumerRecord":{"Topic":"lookups_topic","Partition":{"Value":0,"IsSpecial":false},"Offset":{"Value":170948,"IsSpecial":false},"TopicPartition":{"Topic":"lookups_topic","Partition":{"Value":0,"IsSpecial":false}},"TopicPartitionOffset":{"Topic":"lookups_topic","Partition":{"Value":0,"IsSpecial":false},"Offset":{"Value":170948,"IsSpecial":false},"TopicPartition":{"Topic":"lookups_topic","Partition":{"Value":0,"IsSpecial":false}}},"Message":{"Key":null,"Value":"RmV0Y2ggZnJvbSBicm9rZXIgMCBmYWlsZWQ6IFVua25vd24gYnJva2VyIGVycm9y","Timestamp":{"Type":0,"UnixTimestampMs":-1,"UtcDateTime":"1969-12-31T23:59:59.999Z"},"Headers":[]},"Key":null,"Value":"RmV0Y2ggZnJvbSBicm9rZXIgMCBmYWlsZWQ6IFVua25vd24gYnJva2VyIGVycm9y","Timestamp":{"Type":0,"UnixTimestampMs":-1,"UtcDateTime":"1969-12-31T23:59:59.999Z"},"Headers":[],"IsPartitionEOF":false},"Error":{"Code":-1,"IsFatal":false,"Reason":"Unknown broker error","IsError":true,"IsLocalError":false,"IsBrokerError":false},"StackTrace":"   at Confluent.Kafka.Consumer`2.Consume(Int32 millisecondsTimeout)\r\n   at Service.Messages.Consumer.Consume() in D:\\CollaborationTool\\CollaborationTool\\CollaborationToolBackEnd\\Service\\Messages\\Consumer.cs:line 65","Message":"Unknown broker error","Data":{},"InnerException":null,"HelpLink":null,"Source":"Confluent.Kafka","HResult":-2146233088}
ErrorJson={"Code":-1,"IsFatal":false,"Reason":"Unknown broker error","IsError":true,"IsLocalError":false,"IsBrokerError":false}

**Reply by edenhill** [reply]
Due to this issue in the dotnet client we're missing some crucial information from the consumer error, unfortunately:
https://github.com/confluentinc/confluent-kafka-dotnet/issues/1788
This makes it hard to track down the origin of the broker error.

There's no errors shown in the provided logs.
